bowtie/_cli.py

```diff
@@ -1684,6 +1708,623 @@ async def info(
         click.echo(json.dumps(output, indent=2))
 
 
+async def download_versions_of(id: ConnectableId) -> Set[str]:
+    progress = Progress(
+        TextColumn("[bold blue]{task.description}"),
+        SpinnerColumn(finished_text=""),
+        BarColumn(bar_width=None),
+        TextColumn(
+            "[progress.percentage]{task.percentage:>3.0f}%",
+        ),
+        "•",
+        DownloadColumn(),
+        "•",
+        TimeElapsedColumn(),
+        console=console.Console(),
+        transient=True,
+    )
+    task = progress.add_task(
+        description=f"Fetching versions of {id}",
+        total=None,
+    )
+
+    async with httpx.AsyncClient(timeout=10) as client:
```

Comment:
You repeat more or less the same logic in 3 places here in retrieving a report via HTTPX -- it seems like you would benefit from introducing a `Report.fetch(url: URL, client: AsyncClient | None = None)`.

Reply:
and what about the progress reporting?